From f5d2c46ea283336708fe1f0730ad532d9edf249e Mon Sep 17 00:00:00 2001
From: Thomas Blume <Thomas.Blume@suse.com>
Date: Fri, 29 Jul 2022 15:42:24 +0200
Subject: [PATCH] fix(network-legacy): wait for multiple interfaces

when netroot is given and multiple interfaces are specified, wait for
all of them
---
 modules.d/35network-legacy/net-genrules.sh | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/modules.d/35network-legacy/net-genrules.sh b/modules.d/35network-legacy/net-genrules.sh
index 7e89f73d..255cee00 100755
--- a/modules.d/35network-legacy/net-genrules.sh
+++ b/modules.d/35network-legacy/net-genrules.sh
@@ -79,7 +79,11 @@ command -v fix_bootif > /dev/null || . /lib/net-lib.sh
     if [ -n "${RAW_IFACES}${IFACES}" ]; then
         echo 'SUBSYSTEM!="net", GOTO="net_end"'
         echo 'ACTION!="add|change|move", GOTO="net_end"'
+        let IFACECOUNT=0
+
         for iface in $IFACES $RAW_IFACES; do
+            #count number of interfaces for interface wait routine
+            let IFACECOUNT++
             case "$iface" in
                 ??:??:??:??:??:??) # MAC address
                     cond="ATTR{address}==\"$iface\""
@@ -102,7 +106,7 @@ command -v fix_bootif > /dev/null || . /lib/net-lib.sh
 
         for iface in $IFACES; do
             if [ "$bootdev" = "$iface" ] || [ "$NEEDNET" = "1" ]; then
-                if [ -n "$netroot" ] && [ -n "$DRACUT_SYSTEMD" ]; then
+                if [ -n "$netroot" ] && [ -n "$DRACUT_SYSTEMD" ] && [ "$IFACECOUNT" -le 1 ]; then
                     echo "systemctl is-active initrd-root-device.target || [ -f /tmp/net.${iface}.did-setup ]"
                 else
                     echo "[ -f /tmp/net.${iface}.did-setup ]"
-- 
2.37.1

